webserver/webserver/src/main/java/io/helidon/webserver/SocketConfiguration.java

```diff
@@ -818,6 +881,24 @@ public Builder config(Config config) {
             return this;
         }
 
+        @Override
+        public Builder addRequestedUriDiscoveryType(RequestedUriDiscoveryType type) {
+            this.requestedUriDiscoveryTypes.add(type);
+            this.requestedUriDiscoveryEnabled = true;
```

Comment:
These side effects are tricky.

If the user does this:
```
builder.requestedUriDiscoveryEnabled(shouldBeEnabled);
builder.addREquestedUriDiscoveryType(x);
```
then we override the developer's explicit setting of `enabled`.

Yes, it might be a bit weird for the developer to _explicitly_ disable this feature and then set the type, but if the settings are coming from variables set elsewhere in the developer's code then the order in which the developer invokes these methods can give different behavior because of our side-effect.

If there are side effects, better to implement them in the `build()` method or in the constructor of the object that's built, once we know all settings are stable.

Reply:
Agreed, I will make enabled a `Boolean` with `null` default and decide in `build`